src/lib/core/data-table/data-table.spec.ts

```diff
@@ -111,13 +110,27 @@ describe('CdkTable', () => {
     // Add data to the table and recreate what the rendered output should be.
     dataSource.addData();
     expect(dataSource.data.length).toBe(initialDataLength + 1); // Make sure data was added
-    fixture.detectChanges();
-    fixture.detectChanges();
 
     let changedTableContent = [headerContent];
     dataSource.data.forEach(rowData => changedTableContent.push([rowData.a, rowData.b, rowData.c]));
     expect(tableElement).toMatchTableContent(changedTableContent);
   });
+
+  it('should be able to dynamically change the columns for header and rows', () => {
+    expect(dataSource.data.length).toBe(3);
+
+    let initialTableContent = [['Column A', 'Column B', 'Column C']];
+    dataSource.data.forEach(rowData => initialTableContent.push([rowData.a, rowData.b, rowData.c]));
+    expect(tableElement).toMatchTableContent(initialTableContent);
```

Comment:
You typically want to avoid having logic in unit tests like this, instead preferring to hard-code the values. I think this all should be
```ts
expect(tableElement).toMatchTableContent([
  ['Column A', 'Column B', 'Column C'],
  [data[0].a, data[0].b, data[0].c],
  [data[1].a, data[1].b, data[2].c],
  [data[2].a, data[2].b, data[1].c],
]);
```
Which is also much easier to visually see what's going on with the test